File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        # creating two variables- one for tracking index and one for tracking length
        i = n - 1
        length = 0

        while s[i] == " ":
            i -= 1

        while s[i] != " " and i >= 0:
            i -= 1
            length += 1
        
        return length


            

File: 0058-length-of-last-word/0058-length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    let n = s.length;
    // Creating two variables - one for tracking index and one for tracking length
    let i = n - 1;
    let length = 0;

    while (s[i] == " "){
        i--;
    }

    while (i >= 0 && s[i] != 0){
        i--;
        length++;
    }

    return length
};